scripts/generate_readme/category_index_generator.py

```diff
@@ -0,0 +1,169 @@
+"""Category index generator for KFP components and pipelines."""
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+from .constants import CATEGORY_README_TEMPLATE
+from .metadata_parser import MetadataParser
+from .utils import format_title
+
+logger = logging.getLogger(__name__)
+
+
+class CategoryIndexGenerator:
+    """Generates category-level README.md that indexes all components/pipelines in a category."""
+    
+    def __init__(self, category_dir: Path, is_component: bool = True):
+        """Initialize the category index generator.
+        
+        Args:
+            category_dir: Path to the category directory (e.g., components/dev/).
+            is_component: True if indexing components, False if indexing pipelines.
+        """
+        self.category_dir = category_dir
+        self.is_component = is_component
+        self.category_name = category_dir.name
+        
+        # Set up Jinja2 environment
+        template_dir = Path(__file__).parent / 'templates'
+        self.env = Environment(
+            loader=FileSystemLoader(template_dir),
+            trim_blocks=True,
+            lstrip_blocks=True,
+        )
+        self.template = self.env.get_template(CATEGORY_README_TEMPLATE)
+    
+    def _find_items_in_category(self) -> List[Path]:
+        """Find all component/pipeline directories within the category.
+        
+        Returns:
+            List of paths to component/pipeline directories.
+        """
+        items = []
+        
+        # Look for subdirectories containing component.py or pipeline.py
+        target_file = 'component.py' if self.is_component else 'pipeline.py'
+        
+        for subdir in self.category_dir.iterdir():
```

Comment:
Is self.category_dir ever validated as existing? Not sure if this gracefully handles that.